tests/test_mark_tokens.py

```diff
@@ -582,13 +587,71 @@ def test_sys_modules(self):
         continue
       m = self.create_mark_checker(source)
 
-      # Astroid has at least two weird bugs involving metaclasses
-      if self.is_astroid_test and 'metaclass=' in m.atok.tree.as_string():
-        continue
-
       m.verify_all_nodes(self)
 
   if six.PY3:
     def test_dict_merge(self):
       m = self.create_mark_checker("{**{}}")
       m.verify_all_nodes(self)
+
+  def parse_snippet(self, text, node):
+    """
+    Returns the parsed AST tree for the given text, handling issues with indentation and newlines
+    when text is really an extracted part of larger code.
+    """
+    # If text is indented, it's a statement, and we need to put in a scope for indents to be valid
+    # (using textwrap.dedent is insufficient because some lines may not indented, e.g. comments or
+    # multiline strings). If text is an expression but has newlines, we parenthesize it to make it
+    # parsable.
+    # For expressions and statements, we add a dummy statement '_' before it because if it's just a
+    # string contained in an astroid.Const or astroid.Expr it will end up in the doc attribute and be
+    # a pain to extract for comparison
+    indented = re.match(r'^[ \t]+\S', text)
+    if indented:
+      return self.module.parse('def dummy():\n' + text).body[0].body[0]
+    if util.is_expr(node):
+      return self.module.parse('_\n(' + text + ')').body[1].value
+    if util.is_module(node):
+      return self.module.parse(text)
+    return self.module.parse('_\n' + text).body[1]
+
+  def test_assert_nodes_equal(self):
+    """
+    Checks that assert_nodes_equal actually fails when given different nodes
+    """
+
+    def check(s1, s2):
+      n1 = self.module.parse(s1)
+      n2 = self.module.parse(s2)
+      with self.assertRaises(AssertionError):
+        self.assert_nodes_equal(n1, n2)
+
+    check('a', 'b')
```

Comment:
How about a helper like `compare(s1, s2)`, and test cases like
```
self.assertEqual(compare('a', 'b'), False)
self.assertEqual(compare('a', 'a'), True)
```
It would be good to know that it works correctly for both positive and negative cases.

Should there be trickier test cases, e.g. one that differs only in the docstring, or those differing in `1.0` vs `1`?

Reply:
I think it working for positive cases is strongly tested by the rest of the tests, particularly via `verify_all_nodes`.

I'll add a few cases as suggested, but let's not get carried away, there's no end to the cases we could come up with and it doesn't all have to be within this PR.